### ISSUE TITLE
[Bug]: DescribeCollection RPC fails in DataNode failover

### ISSUE THREAD

**Issue by letian-jiang** [milvus-io/milvus#15823]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: latest
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

DataNode and DataCoord crashed before collection dropped. In failover, DescribeCollection RPCs from DataNode to RootCoord fail and retry endlessly. 

RootCoord Log
```
[2022/03/02 15:10:43.938 +08:00] [DEBUG] [root_coord.go:1376] [DescribeCollection] [role=rootcoord] ["collection name"=] [msgID=0]
[2022/03/02 15:10:43.938 +08:00] [ERROR] [root_coord.go:1388] ["DescribeCollection failed"] [role=rootcoord] ["collection name"=] [msgID=0] [error="can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).DescribeCollection\n\t/Users/zilliz/milvus/internal/rootcoord/root_coord.go:1388\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).DescribeCollection\n\t/Users/zilliz/milvus/internal/distributed/rootcoord/service.go:354\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler.func1\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:915\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/Users/zilliz/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:917\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]
[2022/03/02 15:10:43.943 +08:00] [DEBUG] [root_coord.go:1376] [DescribeCollection] [role=rootcoord] ["collection name"=] [msgID=0]
[2022/03/02 15:10:43.943 +08:00] [ERROR] [root_coord.go:1388] ["DescribeCollection failed"] [role=rootcoord] ["collection name"=] [msgID=0] [error="can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).DescribeCollection\n\t/Users/zilliz/milvus/internal/rootcoord/root_coord.go:1388\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).DescribeCollection\n\t/Users/zilliz/milvus/internal/distributed/rootcoord/service.go:354\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler.func1\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:915\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/Users/zilliz/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:917\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]
[2022/03/02 15:11:03.957 +08:00] [DEBUG] [root_coord.go:1376] [DescribeCollection] [role=rootcoord] ["collection name"=] [msgID=0]
[2022/03/02 15:11:03.957 +08:00] [ERROR] [root_coord.go:1388] ["DescribeCollection failed"] [role=rootcoord] ["collection name"=] [msgID=0] [error="can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/rootcoord.(*Core).DescribeCollection\n\t/Users/zilliz/milvus/internal/rootcoord/root_coord.go:1388\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).DescribeCollection\n\t/Users/zilliz/milvus/internal/distributed/rootcoord/service.go:354\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler.func1\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:915\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/Users/zilliz/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/rootcoordpb._RootCoord_DescribeCollection_Handler\n\t/Users/zilliz/milvus/internal/proto/rootcoordpb/root_coord.pb.go:917\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/Users/zilliz/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934"]
```

DataNode Log
```
2022/03/02 15:10:23.915 +08:00] [DEBUG] [flow_graph_manager.go:39] ["received Vchannel Info"] [vChannelName=by-dev-rootcoord-dml_0_431542760693501249v0] ["Unflushed Segment Number"=0] ["Flushed Segment Number"=1]
[2022/03/02 15:10:23.915 +08:00] [DEBUG] [flow_graph_manager.go:39] ["received Vchannel Info"] [vChannelName=by-dev-rootcoord-dml_1_431542760693501249v1] ["Unflushed Segment Number"=0] ["Flushed Segment Number"=1]
[2022/03/02 15:10:23.920 +08:00] [DEBUG] [minio_kv.go:96] ["MinioKV new MinioKV success."]
[2022/03/02 15:10:23.920 +08:00] [INFO] [data_sync_service.go:190] ["Recover Segment NumOfRows form checkpoints"] [InsertChannel=by-dev-rootcoord-dml_1_431542760693501249v1] [SegmentID=431542760798158850] [NumOfRows=1497]
[2022/03/02 15:10:23.920 +08:00] [DEBUG] [segment_replica.go:382] ["Add Flushed segment"] ["segment ID"=431542760798158850] ["collection ID"=431542760693501249] ["partition ID"=431542760693501250] ["channel name"=by-dev-rootcoord-dml_1_431542760693501249v1]
[2022/03/02 15:10:23.920 +08:00] [INFO] [segment_replica.go:419] ["statsBinlogs is empty"]
[2022/03/02 15:10:23.922 +08:00] [DEBUG] [minio_kv.go:96] ["MinioKV new MinioKV success."]
[2022/03/02 15:10:23.922 +08:00] [INFO] [data_sync_service.go:190] ["Recover Segment NumOfRows form checkpoints"] [InsertChannel=by-dev-rootcoord-dml_0_431542760693501249v0] [SegmentID=431542760798158849] [NumOfRows=1503]
[2022/03/02 15:10:23.922 +08:00] [DEBUG] [segment_replica.go:382] ["Add Flushed segment"] ["segment ID"=431542760798158849] ["collection ID"=431542760693501249] ["partition ID"=431542760693501250] ["channel name"=by-dev-rootcoord-dml_0_431542760693501249v0]
[2022/03/02 15:10:23.922 +08:00] [INFO] [segment_replica.go:419] ["statsBinlogs is empty"]
[2022/03/02 15:10:23.922 +08:00] [ERROR] [segment_replica.go:673] ["Grpc error"] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).getCollectionSchema\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:673\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).initPKBloomFilter\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:421\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).addFlushedSegment\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:402\ngithub.com/milvus-io/milvus/internal/datanode.(*dataSyncService).initNodes\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:195\ngithub.com/milvus-io/milvus/internal/datanode.newDataSyncService\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:90\ngithub.com/milvus-io/milvus/internal/datanode.(*flowgraphManager).addAndStart\n\t/Users/zilliz/milvus/internal/datanode/flow_graph_manager.go:58\ngithub.com/milvus-io/milvus/internal/datanode.(*DataNode).handlePutEvent\n\t/Users/zilliz/milvus/internal/datanode/data_node.go:373\ngithub.com/milvus-io/milvus/internal/datanode.(*channelEventManager).Run.func1\n\t/Users/zilliz/milvus/internal/datanode/event_manager.go:56"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [flow_graph_manager.go:60] ["new data sync service fail"] [vChannelName=by-dev-rootcoord-dml_1_431542760693501249v1] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
[2022/03/02 15:10:23.922 +08:00] [ERROR] [segment_replica.go:673] ["Grpc error"] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"] [stack="github.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).getCollectionSchema\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:673\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).initPKBloomFilter\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:421\ngithub.com/milvus-io/milvus/internal/datanode.(*SegmentReplica).addFlushedSegment\n\t/Users/zilliz/milvus/internal/datanode/segment_replica.go:402\ngithub.com/milvus-io/milvus/internal/datanode.(*dataSyncService).initNodes\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:195\ngithub.com/milvus-io/milvus/internal/datanode.newDataSyncService\n\t/Users/zilliz/milvus/internal/datanode/data_sync_service.go:90\ngithub.com/milvus-io/milvus/internal/datanode.(*flowgraphManager).addAndStart\n\t/Users/zilliz/milvus/internal/datanode/flow_graph_manager.go:58\ngithub.com/milvus-io/milvus/internal/datanode.(*DataNode).handlePutEvent\n\t/Users/zilliz/milvus/internal/datanode/data_node.go:373\ngithub.com/milvus-io/milvus/internal/datanode.(*channelEventManager).Run.func1\n\t/Users/zilliz/milvus/internal/datanode/event_manager.go:56"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [event_manager.go:58] ["handle put event fail, starting retry"] [vChanName=by-dev-rootcoord-dml_1_431542760693501249v1] ["retry interval"=20s] [error="fail to add and start flowgraph for vChanName: by-dev-rootcoord-dml_1_431542760693501249v1, err: describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [flow_graph_manager.go:60] ["new data sync service fail"] [vChannelName=by-dev-rootcoord-dml_0_431542760693501249v0] [error="describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
[2022/03/02 15:10:23.922 +08:00] [WARN] [event_manager.go:58] ["handle put event fail, starting retry"] [vChanName=by-dev-rootcoord-dml_0_431542760693501249v0] ["retry interval"=20s] [error="fail to add and start flowgraph for vChanName: by-dev-rootcoord-dml_0_431542760693501249v0, err: describe collection 431542760693501249 from rootcoord wrong: DescribeCollection failed: can't find collection id : 431542760693501249"]
```

### Anything else?

Extracted from #15608

**Comment by yanliang567** [comment]
/assign @letian-jiang 
i think you are working on this issue.

/unassign

**Comment by DrBamba** [comment]
Is this issue solved in 2.0.1? If not is there an expected estimation as for when will it be solved?

**Comment by letian-jiang** [comment]
> Is this issue solved in 2.0.1? If not is there an expected estimation as for when will it be solved?

Solved in 2.0.2
